File: app/src/main/java/com/lcl/lclmeasurementtool/database/Entity/ConnectivityDAO.java
package com.lcl.lclmeasurementtool.database.Entity;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.TypeConverters;
import androidx.room.Update;

import com.lcl.lclmeasurementtool.Utils.LocationUtils;

import java.util.List;

/**
 * A DAO for connectivity measurement
 */
@Dao
@TypeConverters({LocationUtils.class})
public interface ConnectivityDAO {

    // CREATE
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    public void insert(Connectivity... connectivities);

    // READ
    @Query("SELECT * FROM connectivity_table")
    public LiveData<List<Connectivity>> retrieveAllConnectivities();


    @Query("SELECT * FROM connectivity_table")
    public List<Connectivity> retrieveAllConnectivitiesSynchronous();


    @Query("SELECT * FROM connectivity_table WHERE time_stamp >= :d1 AND time_stamp <= :d2")
    public LiveData<List<Connectivity>> retrieveConnectivitiesBetweenDates(String d1, String d2);

    // UPDATE
    @Update
    public void updateConnectivity(Connectivity... connectivities);

    // DELETE
    @Delete
    public void deleteConnectivity(Connectivity... connectivities);

    @Query("DELETE FROM connectivity_table")
    public void deleteAll();

    @Query("DELETE FROM connectivity_table WHERE time_stamp >= :d1 AND time_stamp <= :d2")
    public void deleteConnectivityStatsBetweenDates(String d1, String d2);
}

File: app/src/main/java/com/lcl/lclmeasurementtool/model/dao/ConnectivityDao.kt
package com.lcl.lclmeasurementtool.model.dao

import androidx.room.*
import com.lcl.lclmeasurementtool.model.datamodel.ConnectivityReportModel

@Dao
interface ConnectivityDao {
    // CREATE
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(vararg connectivity: ConnectivityReportModel)

    // READ
    @Query("SELECT * FROM connectivity_table ORDER BY time_stamp ASC")
    fun getAll(): List<ConnectivityReportModel>

    // DELETE
    @Query("DELETE FROM connectivity_table")
    fun deleteAll()
}